### DISCUSSION TITLE
Response Callback

### DISCUSSION THREAD

**Discussion by DRKolev-code** [CopilotC-Nvim/CopilotChat.nvim#313] [Q&A] [answered]
I am trying to save the chat in a file after each response. For some reason when `CopiltChatSave` is present in the callback function the callback function is only called once in a minute or so.

I can manually spam `: CopilotCatSave test` and it works every time. What am I missing with the callback approach?

```
callback = function()
    require("CopilotChat").save("test")
    print("Chat saved to: test")
end, -- Callback to use when ask response is received
```

**Comment by deathbeam** [comment, marked as answer]
@DRKolev-code hmm I just tried it and it works perfectly fine for me, did you checked logs as well? It shoudl show like saved to bla bla

**Reply by DRKolev-code** [reply]
My fault, sorry for wasting your time. I had placed a filter for messages that are related to saving in my `noice` config. Thanks for checking. I think I was seeing my print statement from inside the `callback` function only once every minute due to race conditions with the print statements from inside the `CopilotChatSave` function.
